README.md

```diff
@@ -1,2 +1,208 @@
-# java-agent
-Agent-based OpenTracing instrumentation in Java
+[![Build Status][ci-img]][ci] [![Released Version][maven-img]][maven]
+
+# Java Agent for OpenTracing
+
+Java Agent for instrumenting Java applications using an OpenTracing compliant Tracer.
+
+The instrumentation is performed in a non-intrusive manner leveraging the [ByteMan project](http://byteman.jboss.org/) to
+define a set of rules. These rules can be used in three ways:
+
+* Directly instrument a technology/framework (e.g. java.net.HttpURLConnection)
+
+* Install a framework integration (e.g. OkHttp, Servlet)
+
+* Define custom rules specific to an application (e.g. create spans to scope important internal units of work,
+or add tags to an existing span to identify business relevant properties)
+
+## Usage
+
+The _Java Agent for OpenTracing_ obtains an OpenTracing compliant Tracer using the
+[Global Tracer](https://github.com/opentracing-contrib/java-globaltracer) project. It makes use of the
+[Span Manager](https://github.com/opentracing-contrib/java-spanmanager) project to manage the
+propagation of active Spans through various frameworks and technologies used by the application - which
+can also be used by the application code to access the current active span to add tags or create its own Spans.
+
+The Java Agent can be used in two ways:
+
+### Prepackaged Agent, Tracer and Dependencies
+
+This approach is to build an uber jar, using the maven assembly or shade plugins, to package together
+the `opentracing-agent.jar`, the OpenTracing compliant `Tracer`, any framework integrations, rules, etc.
+
+The important point to remember is that, because the resulting jar will still be used as a javaagent, it needs the
+manifest entries copied from the manifest in the `opentracing-agent.jar`.
+
+This approach is useful when wanting to instrument applications where modification of the classpath is not
+possible (e.g. executable jars), or wanting to maintain separation between the application and the tracing
+mechanism. It also benefits from being able to selective choose the framework integrations (and versions) that
+are required.
+
+NOTE: An [issue](https://github.com/opentracing-contrib/java-agent/issues/3) has been created to discuss providing tool support for this option.
+
+
+### Tracer and Framework Integrations on Classpath
+
+This approach uses the plain `opentracing-agent.jar` provided by this project, and obtains the OpenTracing
+Tracer and any required framework integrations, rules, etc. from the classpath.
+
+```java
+java -javaagent path/to/opentracing-agent.jar ...
+```
+
+#### Spring Boot Example
```

Comment:
Would prefer at this stage to keep the repo independent of any particular tracer, but could add a section with links to examples, where we could build up a list of tracer specific examples/docs. I have a simple example repo for hawkular, which I'll add once cleaned up :)